handler/area_types.go

```diff
@@ -0,0 +1,54 @@
+package handler
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-cantabular-dimension-api/contract"
+	"github.com/ONSdigital/dp-cantabular-dimension-api/model"
+	"github.com/gorilla/schema"
+	"github.com/pkg/errors"
+)
+
+// AreaTypes handles requests to /area-types
+type AreaTypes struct {
+	respond responder
+	ctblr   cantabularClient
+}
+
+// NewAreaTypes returns a new AreaTypes handler
+func NewAreaTypes(r responder, c cantabularClient) *AreaTypes {
+	return &AreaTypes{
+		respond: r,
+		ctblr:   c,
+	}
+}
+
+// Get is the handler for GET /area-types
+func (at *AreaTypes) Get(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var req contract.GetAreaTypesRequest
+	if err := schema.NewDecoder().Decode(&req, r.URL.Query()); err != nil {
+		at.respond.Error(ctx, w, errors.Wrap(err, "failed to decode query parameter from get area-types request"))
```

Comment:
Could probably return status 400 here. I actually just added a ErrorWithStatus function to the PR I'm working on if you didn't want to create a new error type just for the status code. https://github.com/ONSdigital/dp-net/pull/31/files#L97. There some unit tests for it too

Reply:
updated to use `dp-net` responder